### DISCUSSION TITLE
Add support for selecting an existing attachment

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5989] [Feature Requests]
It would be very nice to have a tool that allowed you to link an existing attachment from another record with a new record.

_Originally posted by @grantfitzsimmons in https://github.com/specify/specify7/issues/84#issuecomment-1666007042_

At this time, attachments can only be linked to a single record from the interface in Specify, though the tables in the database are configured so that a single attachment can be linked to multiple records.

e.g. `Collection Object -> Collection Object Attachment -> Attachment`

Multiple `Collection Object Attachment` records can link to the same `Attachment`, so adding an interface to search existing attachments and make the join in the UI would be desirable. 
            
**Requested By:** Several users (KU Invertebrate Paleontology, KU Ichthyology, University of Copenhagen, and others)

**Comment by maxpatiiuk** [comment]
This would be really nice, and not complicated on the front-end.
The back-end might be tricker though - we might have to adjust the cascade deletion rule for attachments to not delete the attachment when related record is deleted as long as attachment is still used in some record

**Comment by grantfitzsimmons** [comment]
> **Shared Attachments**
> The community voiced the need for attachments to connect to multiple specimens or other data tables:
> **Cody Thompson**: “It would be great if we can... really hone in on the ability to say, take an image of a mouse, and the mouse has a parasite.”
> **Nate** **Shoobs**: “We have scans of our collection ledgers that are linked to groups of 25 COs.”
> **Natalia López Carranza:** “YES!”

From our ["Open Discussion on Associated Media Management in Specify 7"](https://discourse.specifysoftware.org/t/open-discussion-on-associated-media-management-in-specify-7/2156/1)

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/open-discussion-on-associated-media-management-in-specify-7/2156/3
